src/classes/RD2_ValidationService_TEST.cls

```diff
@@ -200,6 +204,257 @@ private with sharing class RD2_ValidationService_TEST {
+            .withAmount(-10)
+            .withInstallmentPeriodYearly()
+            .build();
+
+        try {
+            insert rd;
+        } catch (Exception e) {
+            errMessage = e.getMessage();
+        }
+        System.assert(errMessage.contains(System.Label.RD2_CheckValidAmountOnRD));
+    }
+
+    /***
+    * @description Verifies if it creates Recurring donations for closed status when
+        no amount is specified.
+    */
+    @isTest
+    private static void shouldCreateRDWhenNoAmountIsSpecifiedForClosedRD() {
+        String errMessage = '';
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact con = UTIL_UnitTestData_TEST.getContact();
+        insert con;
+
+        npe03__Recurring_Donation__c rd = new TEST_RecurringDonationBuilder()
+            .withContact(con.Id)
+            .withInstallmentPeriodYearly()
+            .withStatusClosed()
+            .build();
+
+        try {
+            insert rd;
+        } catch (Exception e) {
+            errMessage = e.getMessage();
+        }
+        System.assert(String.isBlank(errMessage), 'Recurring Donation should be created successfully: ' + errMessage);
+    }
+
+    /***
+    * @description Verifies if it creates Recurring donations for closed status when
+        zero amount is specified.
+    */
+    @isTest
+    private static void shouldCreateRDWhenZeroAmountIsSpecifiedForClosedRD() {
+        String errMessage = '';
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact con = UTIL_UnitTestData_TEST.getContact();
+        insert con;
+
+        npe03__Recurring_Donation__c rd = new TEST_RecurringDonationBuilder()
+            .withContact(con.Id)
+            .withInstallmentPeriodYearly()
+            .withAmount(0)
+            .withStatusClosed()
+            .build();
+
+        try {
+            insert rd;
+        } catch (Exception e) {
+            errMessage = e.getMessage();
+        }
+        System.assert(String.isBlank(errMessage), 'Recurring Donation should be created successfully: ' + errMessage);
+    }
+
+    /***
+    * @description Verifies if it displays error message when installment period
+        is monthly and day of month is blank
+    */
+    @isTest
+    private static void shouldFailWhenDayOfMonthIsBlankForMonthlyInstallmentPeriod() {
+        String errMessage = '';
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact con = UTIL_UnitTestData_TEST.getContact();
+        insert con;
+
+        npe03__Recurring_Donation__c rd = new TEST_RecurringDonationBuilder()
+            .withContact(con.Id)
+            .withAmount(10)
+            .withInstallmentPeriodMonthly()
+            .build();
+
+        try {
+            insert rd;
+        } catch (Exception e) {
+            errMessage = e.getMessage();
+        }
+        String expectedMessage = String.format(
+            System.Label.RD2_CheckValidDayOfMonth,
```

Comment:
Should the error be `RD2_DayOfMonthIsReqForMonthlyInstallments` when "Day of Month" is not provided for Monthly "Installment Period"?